src/main/java/org/eclipse/uprotocol/uri/validator/UriValidator.java

```diff
@@ -192,6 +192,21 @@ static boolean matches(UUri uriToMatch, UUri candidateUri) {
                 matchesEntity(uriToMatch, candidateUri) &&
                 matchesResource(uriToMatch, candidateUri);
     }
+
+
+    /**
+     * Checks if the URI has a wildcard in any of its fields.
+     *
+     * @param uri The URI to check for wildcards.
+     * @return True if the URI has a wildcard, False otherwise.
+     */
+    static boolean hasWildcard(UUri uri) {
+        return !isEmpty(uri) && 
+                (uri.getAuthorityName().equals(UriFactory.WILDCARD_AUTHORITY) ||
+                uri.getUeId() == UriFactory.WILDCARD_ENTITY_ID ||
```

Comment:
I think we should use bitwise AND operation here as FFFF, 1FFFF etc all are wildcard.

```suggestion
               (uri.getUeId() & UriFactory.WILDCARD_ENTITY_ID)== UriFactory.WILDCARD_ENTITY_ID ||
```